UnityProject/Assets/Scripts/Managers/GameManager.cs

```diff
@@ -467,10 +467,10 @@ public void EndRound()
 			}
 
 			CurrentRoundState = RoundState.Ended;
+			GameMode.EndRound();
 			EventManager.Broadcast(Event.RoundEnded, true);
 			counting = false;
 
-			GameMode.EndRound();
```

Comment:
This sort of change might have unintended consequences. Are you sure there's no code that depends on the event to be broadcast _before_ the round actually ends?